### DISCUSSION TITLE
How to validate all elements in a field array? #useFieldArray

### DISCUSSION THREAD

**Discussion by rameshx** [react-hook-form/react-hook-form#5913] [Q&A]
What's the best way to trigger the validation of all elements in a field array when one of the values change?

I want to make sure that the values of all elements are unique. The custom validate function is working but when the duplicate value is corrected the validation is not triggered for the rest of the elements.

code sandbox: https://codesandbox.io/s/react-hook-form-usefieldarray-virtual-inputs-forked-sr5qz?file=/src/App.js

Steps to reproduce:
1. modify the values so that there are duplicates
2. click submit, now the ugly red border will be applied to inputs with non-unique values.
3. change one of the values to make it unique
4. the red border is only removed for the input whose value is changed

how I can utilize the `trigger` function to validate the rest of the inputs? 

PS: clicking on the submit again is validating all inputs but since the reValidation happens on onChange, I want this behaviour for all fields.

**Comment by bluebill1049** [comment]
Do you mean something like `trigger('fiedlArray')`? https://github.com/react-hook-form/react-hook-form/issues/2381

**Reply by rameshx** [reply]
@bluebill1049 Yes, but where should I call `trigger('fieldArray')` in my case? I tried something but it's making the page unresponsive.

**Reply by bluebill1049** [reply]
this feature is still under feature request. 

> unresponsive

what's causing this problem? trigger?

**Reply by rameshx** [reply]
Yes, I tried calling it in useEffect with fields as a dependency.
Is there some manual approach that I can use for now?
